rest/config.go

```diff
@@ -1344,26 +1348,31 @@ func (sc *StartupConfig) Validate(ctx context.Context, isEnterpriseEdition bool)
 
 // SetupServerContext creates a new ServerContext given its configuration and performs the context validation.
 func SetupServerContext(ctx context.Context, config *StartupConfig, persistentConfig bool) (*ServerContext, error) {
-	// Logging config will now have been loaded from command line
-	// or from a sync_gateway config file so we can validate the
-	// configuration and setup logging now
-	if err := config.SetupAndValidateLogging(ctx); err != nil {
-		// If we didn't set up logging correctly, we *probably* can't log via normal means...
-		// as a best-effort, last-ditch attempt, we'll log to stderr as well.
-		log.Printf("[ERR] Error setting up logging: %v", err)
-		return nil, fmt.Errorf("error setting up logging: %v", err)
+	// If SetupServerContext is called while any other go routines that might use logging are running, it will
+	// cause a data race, therefore only initialize logging and other globals on the first call. From a main
+	// program, there is only one ServerContext.
+	if serverContextGlobalsInitialized.CompareAndSwap(false, true) {
```

Comment:
I'm not sure how this is intended to work for a suite of tests that only invoke `SetupServerContext` once per test.

Is it expected that only the first test gets to set global config, and all others can't?

I'd have expected this flag to be reset between tests maybe (`t.Cleanup`?), if the intention is to only allow this to be run once _per test_ to avoid races